linetools/isgm/utils.py

```diff
@@ -416,6 +434,26 @@ def table_from_complist(complist):
     assert len(key_order) == len(tab.keys())
     tab = tab[key_order]
 
+    # May need to add an HI component here as done in the method below
+    if NHI_obj is not None:
+        # Existing row in Table?
+        mt = np.where((tab['Z'] == 1) & (tab['ion']==1))[0]
+        if len(mt) == 1:
+            tab[mt[0]]['logN'] = NHI_obj.NHI
+            tab[mt[0]]['sig_logN'] = np.mean(NHI_obj.sig_NHI) # Allow for two values
+            tab[mt[0]]['flag_N'] = NHI_obj.flag_NHI
+        else: # Add a dummy row
+            tab.add_row(tab[0])
+            tab['logN'][-1] = NHI_obj.NHI
+            tab['sig_logN'][-1] = np.mean(NHI_obj.sig_NHI) # Allow for two values
+            tab['flag_N'][-1] = NHI_obj.flag_NHI
+            #
+            tab['Z'][-1] = 1
+            tab['ion'][-1] = 1
+            tab['Ej'][-1] = 0.
+            tab['ion_name'][-1] = 'HI'
+            tab['comp_name'][-1] = 'HI_z{:0.5f}'.format(tab['z_comp'][-1])
```

Comment:
Do not understand the need of a NHI object here, the idea here is to represent a list of AbsComponent relevant info into a table.

Reply:
Don't worry about it...